File: clients/java/src/main/java/com/priostack/acn/SpaceResult.java
package com.priostack.acn;

import com.fasterxml.jackson.databind.JsonNode;

/** Result of {@link ACNClient#createSpace}. Use {@link #spaceId} for writes. */
public final class SpaceResult {

    private final String spaceId;
    private final String accountId;
    private final String persistenceMode;
    private final String protectionLevel;
    private final JsonNode raw;

    public SpaceResult(
            String spaceId,
            String accountId,
            String persistenceMode,
            String protectionLevel,
            JsonNode raw) {
        this.spaceId = spaceId;
        this.accountId = accountId;
        this.persistenceMode = persistenceMode;
        this.protectionLevel = protectionLevel;
        this.raw = raw;
    }

    /** The resolved space id; pass this to {@code store}/{@code fetch}/{@code grant}. */
    public String spaceId() {
        return spaceId;
    }

    public String accountId() {
        return accountId;
    }

    public String persistenceMode() {
        return persistenceMode;
    }

    public String protectionLevel() {
        return protectionLevel;
    }

    public JsonNode raw() {
        return raw;
    }

    @Override
    public String toString() {
        return "SpaceResult{spaceId=" + spaceId + ", persistenceMode=" + persistenceMode + '}';
    }
}

File: clients/php/src/Result/SpaceResult.php
<?php

declare(strict_types=1);

namespace Priostack\Acn\Result;

/**
 * Result of {@see \Priostack\Acn\Client::createSpace()}.
 *
 * Use the returned {@see $spaceId} for subsequent {@code store}/{@code fetch}
 * calls — it is the server-resolved id, not necessarily the display name.
 */
final class SpaceResult
{
    /**
     * @param array<string, mixed> $raw
     */
    public function __construct(
        public readonly string $spaceId,
        public readonly string $accountId,
        public readonly string $persistenceMode,
        public readonly string $protectionLevel,
        public readonly array $raw = [],
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromData(array $data): self
    {
        return new self(
            spaceId: (string) ($data['spaceId'] ?? ''),
            accountId: (string) ($data['accountId'] ?? ''),
            persistenceMode: (string) ($data['persistenceMode'] ?? ''),
            protectionLevel: (string) ($data['protectionLevel'] ?? ''),
            raw: $data,
        );
    }
}
